daceml/transformation/constant_folding.py

```diff
@@ -191,37 +196,36 @@ def apply(self, state: dace.SDFGState, sdfg: dace.SDFG):
         remove_node_and_computation(sdfg, state, node)
 
 
-def remove_node_and_computation(sdfg: dace.SDFG, state: dace.SDFGState,
-                                node: nd.Node):
+def remove_node_and_computation(sdfg: dace.SDFG,
+                                state: dace.SDFGState,
+                                node: nd.Node,
+                                connector: Optional[str] = None):
     """ Remove a node and the parent nodes that compute this node, if the outputs are not used elsewhere.
 
         :param sdfg: the sdfg containing the node.
         :param state: the state containing the node.
         :param node: the node to remove
+        :param connector: if not None, the computation of the connector of
+                          ``node`` will be removed, but not ``node`` itself.
     """
-    queue = deque([node])
-    while len(queue) > 0:
-        current_node = queue.popleft()
-
-        edges = state.in_edges(current_node)
-        state.remove_node(current_node)
+    if connector is not None:
+        if connector not in node.in_connectors:
+            return
+        node.remove_in_connector(connector)
+        edges = state.in_edges_by_connector(node, connector)
+        for e in edges:
+            state.remove_edge(e)
+    else:
+        edges = state.out_edges(node)
         for e in edges:
-            next_node = e.src
-            data_used_in_other_states = isinstance(next_node, nd.AccessNode) and \
-                                        any(n.data == next_node.data
-                                            for s in sdfg.nodes()
-                                            for n in s.nodes() if s is not state)
-
-            if len(state.out_edges(
-                    next_node)) == 0 and not data_used_in_other_states:
-                queue.append(next_node)
-
-    # remove all now useless data descriptors
-    all_read_or_written_data = set(e.data.data for s in sdfg.nodes()
-                                   for e in s.edges())
-    all_read_or_written_data = all_read_or_written_data.union(
-        node.data for node, _ in sdfg.all_nodes_recursive()
-        if isinstance(node, nd.AccessNode))
-    to_delete = set(sdfg.arrays) - all_read_or_written_data
-    for name in to_delete:
-        del sdfg.arrays[name]
+            state.remove_edge(e)
+
+    pass_pipeline.Pipeline([
+        dead_dataflow_elimination.DeadDataflowElimination()
```

Comment:
interesting use. won't it remove much more than that single node's dependencies?

Reply:
Potentially, but in the NN sdfgs there's usually no dead code.
We can fix it later if we run into issues